cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c

```diff
@@ -0,0 +1,649 @@
+            dev->cb(dev, IEEE802154_RADIO_INDICATION_RX_DONE);
+        }
+        /* If all failed, simply drop the frame and continue listening to
+         * incoming frames */
+        else {
+            entry->status = RFC_DATA_ENTRY_PENDING;
+        }
+
+    }
+
+    if ((RFC_DBELL->RFCPEIFG & CPE_IRQ_LAST_COMMAND_DONE) &&
+        (RFC_DBELL->RFCPEIEN & CPE_IRQ_LAST_COMMAND_DONE)) {
+        RFC_DBELL_NONBUF->RFCPEIFG = ~CPE_IRQ_LAST_COMMAND_DONE;
+        RFC_DBELL->RFCPEIEN &= ~CPE_IRQ_LAST_COMMAND_DONE;
+
+        if (rf_cmd_prop_tx_adv.status == RFC_PROP_DONE_OK) {
+            rf_cmd_prop_tx_adv.status = RFC_IDLE;
+
+            switch (_tx_state) {
+            case STATE_TX:
+                _tx_state = TX_STATE_IDLE;
+                _set_ifs_timer(cc26x2_cc13x2_rf_tx_psdu_len() > IEEE802154_SIFS_MAX_FRAME_SIZE);
+                _tx_done++;
+                dev->cb(dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
+                break;
+            case STATE_ACK:
+                _tx_state = TX_STATE_IDLE;
+                /* restore RX */
+                cc26x2_cc13x2_rf_rx_start();
+
+                _set_ifs_timer(false);
+                dev->cb(dev, IEEE802154_RADIO_INDICATION_RX_DONE);
+                break;
+            default:
+                DEBUG("cc26x2_cc13x2_rf]: undefined TX state\n");
+                break;
+            }
+        }
+
+        if (rf_cmd_prop_cs.status == RFC_PROP_DONE_IDLE ||
+            rf_cmd_prop_cs.status == RFC_PROP_DONE_BUSY) {
+            _cca_result =
+                rf_cmd_prop_cs.status == RFC_PROP_DONE_IDLE ? CCA_RESULT_IDLE :
+                                                              CCA_RESULT_BUSY;
+            rf_cmd_prop_cs.status = RFC_IDLE;
+
+            dev->cb(dev, IEEE802154_RADIO_CONFIRM_CCA);
+        }
+    }
+}
+
+static int _write(ieee802154_dev_t *dev, const iolist_t *iolist)
+{
+    (void)dev;
+
+    return cc26x2_cc13x2_rf_write(iolist);
+}
+
+static int _read(ieee802154_dev_t *dev, void *buf, size_t max_size,
+                 ieee802154_rx_info_t *info)
+{
+    (void)dev;
+    rfc_data_entry_general_t *entry;
+    uint8_t *psdu;
+    uint16_t pktlen;
+
+    /* shouldn't be null as we informed about a packet */
+    entry = cc26x2_cc13x2_rf_recv();
+    if (entry == NULL) {
+        return -ENOENT;
+    }
+
+    /* first two bytes are PSDU length */
+    pktlen = *(uint16_t *)&entry->data;
+    if (pktlen < CC26X2_CC13X2_RF_RX_STATUS_LEN ||
+        pktlen > IEEE802154G_FRAME_LEN_MAX + CC26X2_CC13X2_RF_RX_STATUS_LEN) {
+        entry->status = RFC_DATA_ENTRY_PENDING;
+        return -ENOENT;
+    }
+    pktlen -= CC26X2_CC13X2_RF_RX_STATUS_LEN;
+    DEBUG("[cc26x2_cc13x2_rf]: reading packet of length %i\n", pktlen);
+
+    psdu = (&entry->data) + sizeof(uint16_t);
+
+    if (max_size < pktlen) {
+        DEBUG("[cc26x2_cc13x2_rf]: buffer is too small\n");
+        entry->status = RFC_DATA_ENTRY_PENDING;
+        return -ENOBUFS;
+    }
```

Comment:
@eduazocar if you could try again here and print before the `if` the `max_size`, `pktlen` and `buf` it would help a lot. It's strange that the caller is passing a buffer too small than the received packet.

(unless a larger than expected frame was received)

Reply:
in the receive process this is returned:
```
2022-10-27 00:19:14,691 # [cc26x2_cc13x2_rf]: reading packet of length 43
2022-10-27 00:19:14,692 # Packet length: 43
2022-10-27 00:19:14,694 # Max size: 43
2022-10-27 00:19:14,710 # Buffer: 0x0 0x0 0x0 0x0 0xE8 0x17 0x0 0x0 0x2D 0x3 0x0 0x0 0x0 0x0 0x0 0x0 0x0 0x0 0x0 0x0 0xD8 0x17 0x0 0x0 0x8 0x0 0x0 0x0 0x0 0x0 0x0 0x0 0x3 0xFE 0x58 0xB8 0x13 0x0 0x0 0x0 0x0 0x0 0x0 
```